lucet-wiggle/generate/src/config.rs

```diff
@@ -3,37 +3,79 @@ use syn::{
     braced,
     parse::{Parse, ParseStream},
     punctuated::Punctuated,
-    Error, Ident, Result, Token,
+    Error, Result, Token,
 };
 use wiggle_generate::config as w;
 
+mod kw {
```

Comment:
You named a number of motivations for this PR: `tracing` and `error transformations` and an update to keep up with cranelift's enhancement's to wasm function type and cranelift signature.  This being the first PR I've reviewed for wiggle, could you please comment on what category of change these `custom_keywords` fall in?

Reply:
Great question! I refactored the parser for the config upstream, and mirrored that refactoring here. Using the custom keywords allows us to use `syn`'s `lookahead` parser-combinator, which means we get better error messages when we come across an unexpected field in the config. The downside is that we have to repeat some of the parsing logic for the various elements from the core config that we are looking for. This isn't the greatest trade-off, but it seems to be the style that `syn` and other parser-combinator users (the `wast` crate exposes very similar primitives) typically do, so hopefully when someone comes across it they say "ah i know whats going on here".